Client/Frontend/Settings/HomepageSettings/WallpaperSettings/v2/WallpaperSettingsViewController.swift

```diff
@@ -0,0 +1,285 @@
+    }
+
+    // MARK: - View setup & lifecycle
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupView()
+        applyTheme()
+        setupNotifications(forObserver: self,
+                           observing: [.DisplayThemeChanged, UIContentSizeCategory.didChangeNotification])
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        applyTheme()
+
+        collectionView.selectItem(at: viewModel.selectedIndexPath, animated: false, scrollPosition: [])
+    }
+
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+
+        if previousTraitCollection?.horizontalSizeClass != traitCollection.horizontalSizeClass
+            || previousTraitCollection?.verticalSizeClass != traitCollection.verticalSizeClass {
+            updateOnRotation()
+        }
+    }
+
+    override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
+        super.viewWillTransition(to: size, with: coordinator)
+        updateOnRotation()
+    }
+}
+
+// MARK: - CollectionView Data Source
+extension WallpaperSettingsViewController: UICollectionViewDelegate, UICollectionViewDataSource {
+
+    func numberOfSections(in collectionView: UICollectionView) -> Int {
+        return viewModel.numberOfSections
+    }
+
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return viewModel.numberOfWallpapers(in: section)
+    }
+
+    func collectionView(_ collectionView: UICollectionView,
+                        viewForSupplementaryElementOfKind kind: String,
+                        at indexPath: IndexPath) -> UICollectionReusableView {
+        guard kind == UICollectionView.elementKindSectionHeader,
+              let headerView = collectionView.dequeueReusableSupplementaryView(
+                ofKind: UICollectionView.elementKindSectionHeader,
+                withReuseIdentifier: WallpaperSettingsHeaderView.cellIdentifier,
+                for: indexPath) as? WallpaperSettingsHeaderView,
+              let headerViewModel = viewModel.sectionHeaderViewModel(for: indexPath.section, dismissView: {
+                  self.dismissView()
+              })
+        else { return UICollectionReusableView() }
+
+        headerView.configure(viewModel: headerViewModel)
+        return headerView
+    }
+
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        guard let cell = collectionView.dequeueReusableCell(
+            withReuseIdentifier: WallpaperCollectionViewCell.cellIdentifier,
+            for: indexPath) as? WallpaperCollectionViewCell,
+                let cellViewModel = viewModel.cellViewModel(for: indexPath)
+        else { return UICollectionViewCell() }
```

Comment:
```suggestion
        guard let cell = collectionView.dequeueReusableCell(
            withReuseIdentifier: WallpaperCollectionViewCell.cellIdentifier,
            for: indexPath) as? WallpaperCollectionViewCell,
                let cellViewModel = viewModel.cellViewModel(for: indexPath)
        else { return UICollectionViewCell() }
```
nit: the second let should be indented equally with the first.